src/github.com/getlantern/flashlight/client/handler.go

```diff
@@ -69,7 +73,11 @@ func (client *Client) intercept(resp http.ResponseWriter, req *http.Request) {
 		return
 	}
 
-	defer connOut.Close()
+	defer func() {
+		if err := connOut.Close(); err != nil {
+			log.Debugf("Error closing the out connection: %s", err)
+		}
+	}()
```

Comment:
I've checked what you mention in the parent comment, and I think it was originally written like this so **intercept** already handles closing connections even if **pipeData** is never reached, due to a **BadGateway** error (i.e. any return statement before reaching **pipeData**). Is this correct?